vyper/semantics/validation/annotation.py

```diff
@@ -182,7 +182,20 @@ def visit_Name(self, node, type_):
         node._metadata["type"] = get_exact_type_from_node(node)
 
     def visit_Subscript(self, node, type_):
-        base_type = get_exact_type_from_node(node.value)
+        if isinstance(node.value, vy_ast.List):
+            base_type = get_possible_types_from_node(node.value)
```

Comment:
to make this clearer, should this be plural?
```suggestion
            base_types = get_possible_types_from_node(node.value)
```

Reply:
Yes, it should be plural for clarity. I will make the changes in a new commit and update once done.

Reply:
I have updated to plural form and also improved the naming to `possible_base_type` and `possible_base_types`. Thanks!